src/cdk-lib/bedrock/README.md

```diff
@@ -1294,3 +1295,143 @@ or alternatively:
+
+TypeScript
+
+```ts
+const cris = bedrock.CrossRegionInferenceProfile.fromConfig({
+  geoRegion: bedrock.CrossRegionInferenceProfileRegion.US,
+  model: bedrock.BedrockFoundationModel.ANTHROPIC_CLAUDE_3_5_SONNET_V2_0
+});
+```
+
+Python
+
+```python
+cris = bedrock.CrossRegionInferenceProfile.from_config(
+  geo_region= bedrock.CrossRegionInferenceProfileRegion.US,
+  model= bedrock.BedrockFoundationModel.ANTHROPIC_CLAUDE_3_5_SONNET_V2_0
+)
+```
+
+## Application inference profile
+
+You can create an application inference profile with one or more Regions to track usage and costs when invoking a model.
+
+To create an application inference profile for one Region, specify a foundation model. Usage and costs for requests made to that Region with that model will be tracked.
+
+To create an application inference profile for multiple Regions, specify a cross region (system-defined) inference profile. The inference profile will route requests to the Regions defined in the cross region (system-defined) inference profile that you choose. Usage and costs for requests made to the Regions in the inference profile will be tracked. You can find the system defined inference profiles by navigating to your console (Amazon Bedrock -> Cross-region inference) or programmatically, for instance using [boto3](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/bedrock/client/list_inference_profiles.html):
+```
+bedrock = session.client("bedrock", region_name="us-east-1")
+bedrock.list_inference_profiles(typeEquals='SYSTEM_DEFINED')
+```
+
+Before using application inference profiles, ensure that:
+- You have appropriate IAM permissions
+- You have access to the models and regions defined in the inference profiles
+- Ensure proper configuration of the required API permissions for inference profile-related actions
+
+Specifically the role you are assuming needs to have permissions for following actions in the IAM policy
+```
+"Action": [
+      "bedrock:GetInferenceProfile",
+      "bedrock:ListInferenceProfiles",
+      "bedrock:DeleteInferenceProfile"
+      "bedrock:TagResource",
+      "bedrock:UntagResource",
+      "bedrock:ListTagsForResource"
+  ]
+```
+You can restrict to specific resources by applying "Resources" tag in the IAM policy.
+```
+"Resource": ["arn:aws:bedrock:*:*:application-inference-profile/*"]
+```
+
+TypeScript
+
+```ts
+// Create an application inference profile for one Region
+// You can use the 'bedrock.BedrockFoundationModel' or pass the arn as a string
+const appInfProfile1 = new ApplicationInferenceProfile(this, 'myapplicationprofile', {
+  inferenceProfileName: 'claude 3 sonnet v1',
+  modelSource: bedrock.BedrockFoundationModel.ANTHROPIC_CLAUDE_SONNET_V1_0,
+  tags: [{key: 'test', value: 'test'}]
+});
+
+// To create an application inference profile across regions, specify the cross region inference profile
+const cris = bedrock.CrossRegionInferenceProfile.fromConfig({
```

Comment:
May be we can update agent documentation on how to set the inference profile..

 ```
 const cris = bedrock.CrossRegionInferenceProfile.fromConfig({
      geoRegion: bedrock.CrossRegionInferenceProfileRegion.US,
      model: bedrock.BedrockFoundationModel.ANTHROPIC_CLAUDE_3_5_HAIKU_V1_0,
    });

    new bedrock.Agent(stack, 'Agent', {
      foundationModel: cris,
      instruction: 'You provide support for developers working with CDK constructs.',
      enableUserInput: true,
    });
  });
```

Reply:
Done in [cf6621f](https://github.com/awslabs/generative-ai-cdk-constructs/pull/800/commits/cf6621fdc9054c3321c404587e403728f80e6eba)